app/javascript/components/widget-controller.jsx

```diff
@@ -1,120 +1,108 @@
 import { mathMod } from 'ramda';
 import FullPanel from './full-panel';
 import SidePanel from './side-panel';
-import Twitter from './twitter';
 import Numbers from './widgets/numbers';
 import Weather from './widgets/weather';
-import Live from './live-stream';
-import lockedIcon from '../../assets/images/locked.svg';
-import unlockedIcon from '../../assets/images/unlocked.svg';
-import { colors } from '../lib/theme';
 
-const IconWrapper = styled.div`
-  position: absolute;
-  left: 20px;
-  top: 20px;
-`;
-
-const Icon = styled.button`
-  height: 21px;
-  width: 16px;
-  border-style: none;
-  background-color: ${colors.black};
-`;
-
-const iconId = 'lock-icon';
+const TWENTYSECONDS = 20000;
 
 const widgets = [
   {
     panel: <Weather.Panel />,
     text: 'Weather',
     children: <Weather.Carousel />,
   },
-  {
-    panel: <Twitter />,
-    text: '@MojoTech',
-    children: 'Last tweeted 2 days ago.',
-  },
   {
     panel: <Numbers.Panel />,
     text: 'MojoTech by the Numbers',
     children: <Numbers.Carousel />,
   },
-  {
-    panel: <Live />,
-    text: 'MojoTech Boulder',
-    children: 'Live.',
-  },
 ];
 
 export class WidgetController extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       index: 0,
-      isLocked: true,
+      time: TWENTYSECONDS,
+      isPaused: 0,
     };
   }
 
   componentDidMount() {
     this.startCarousel();
   }
 
-  switchLock = () => {
-    this.setState(prevState => ({ isLocked: !prevState.isLocked }));
-  };
+  componentWillUnmount() {
+    if (this.intervalID) {
+      clearInterval(this.intervalID);
+    }
+  }
 
   switchPages = ({ keyCode }) => {
     const upKey = 38;
     const downKey = 40;
+    const spacebar = 32;
     if (keyCode === downKey) {
       this.moveDown();
     }
     if (keyCode === upKey) {
       this.moveUp();
     }
+    if (keyCode === spacebar) {
+      this.pause();
+    }
   };
 
   switchToPage = index => {
-    this.setState({ index });
+    this.setState({ index, time: TWENTYSECONDS });
   };
 
   handleClicks = ({ target }) => {
     const targetId = target.id;
-    if (targetId === iconId) {
-      this.switchLock();
-    } else if (targetId.startsWith('widget_')) {
+    if (targetId.startsWith('widget_')) {
       const widgetId = parseInt(targetId.replace('widget_', ''), 10);
       this.switchToPage(widgetId);
     }
   };
 
   startCarousel() {
-    setInterval(() => {
-      if (!this.state.isLocked) {
+    this.intervalID = setInterval(() => {
+      if (!this.state.isPaused) {
+        this.setState(prevState => ({ time: prevState.time - 50 }));
+      }
+      if (this.state.time <= 0) {
         this.moveDown();
       }
-    }, 30000);
+    }, 50);
   }
 
   moveUp() {
+    this.setState({ time: TWENTYSECONDS });
     this.setState(({ index }) => ({
       index: mathMod(index - 1, widgets.length),
     }));
   }
 
   moveDown() {
+    this.setState({ time: TWENTYSECONDS });
     this.setState(({ index }) => ({
```

Comment:
Let's merge these two setStates together `this.setState(({ index, time: TWENTYSECONDS }) => ({...`